examples/transmission_main_example.py

```diff
@@ -47,40 +47,55 @@ def main(args):
             32e9, 0.15, Power(1e-3, 0, 0)) for f in range(1,97)))
 
     trx = [n for n in network.nodes() if isinstance(n, Transceiver)]
-    source, sink = trx[0], trx[1]
- 
-    path = dijkstra_path(network, source, sink)
-    print(f'There are {len(path)} network elements between {source} and {sink}')
-
-    for el in path:
-        si = el(si)
-        print(el)
-
-    nodelist = [n for n in network.nodes() if isinstance(n, (Transceiver, Fiber))]
-    pathnodes = [n for n in path if isinstance(n, (Transceiver, Fiber))]
-    edgelist = [(u, v) for u, v in zip(pathnodes, pathnodes[1:])]
-    node_color = ['#ff0000' if n is source or n is sink else
-                  '#900000' if n in path else '#ffdfdf'
-                  for n in nodelist]
-    edge_color = ['#ff9090' if u in path and v in path else '#ababab'
-                  for u, v in edgelist]
-    labels = {n: n.location.city if isinstance(n, Transceiver) else ''
-              for n in pathnodes}
-
-    fig = figure()
-    pos = {n: (n.lng, n.lat) for n in nodelist}
-    kwargs = {'figure': fig, 'pos': pos}
-    plot = draw_networkx_nodes(network, nodelist=nodelist, node_color=node_color, **kwargs)
-    draw_networkx_edges(network, edgelist=edgelist, edge_color=edge_color, **kwargs)
-    draw_networkx_labels(network, labels=labels, font_size=14, **kwargs)
-    title(f'Propagating from {source.loc.city} to {sink.loc.city}')
-    axis('off')
-    show()
+    if args.list>=1:
+        for el in trx:
+            print(el.uid[4:])
+    else:
+        source, sink = trx[0], trx[1]
+        for el in trx:
```

Comment:
You dont account for the possibility that the user did not supply a source and sink agument, other than by specifying an invalid default value `""`.

Either:
- make these parameters completely optional and dynamically determine a reasonable default value (which is not particularly meaningful,)
- make these values mandatory.

Reply:
Additionally, you also don't account for the possibility that the source and sink are invalid.

For now, it's okay that you're assuming that the node uids encode the network element type in the first four characters, but that may change later (in which case we'll have to fix this code.)

I would break this into two separate loops which `break` once they have identified the correct element, plus a check at the end.

You could write:
```
source = None
for el in trx:
    if args.source == el.uid[4:]:
        source = el
        break
if source is None:
    raise ValueError(f"invalid source node specified: {args.source!r}")
```

You could simplify to:
```
source = next((el for el in trx if args.source == el.uid[4:]), None)
if source is None:
    raise ValueError(f"invalid source node specified: {args.source!r}")
```

Or even:
```
try:
    source = next(el for el in trx if args.source == el.uid[4:])
except StopIteration as e:
    raise ValueError(f"invalid source node specified: {args.source!r}") from e
```

Reply:
Also, try to keep the most common code path at the lowest level of indentation.

As you can see, by intenting the block below, the diff for this PR is much larger than it should be.